src/Route.php

```diff
@@ -3,34 +3,31 @@
 namespace Yiisoft\Router;
 
 use InvalidArgumentException;
-use LogicException;
 use Yiisoft\Http\Method;
 use Yiisoft\Router\Middleware\Callback;
-use Psr\Http\Message\ResponseInterface;
-use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface;
 use Psr\Http\Server\RequestHandlerInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Message\ResponseInterface;
 
 /**
  * Route defines a mapping from URL to callback / name and vice versa
  */
-final class Route implements MiddlewareInterface, RequestHandlerInterface
+final class Route implements MiddlewareInterface
 {
     private ?string $name = null;
     /** @var string[] */
     private array $methods;
     private string $pattern;
     private ?string $host = null;
+    private ?RequestHandlerInterface $stack = null;
```

Comment:
Is it possible to do it without wrapping same way as https://github.com/yiisoft/yii-web/blob/master/src/MiddlewareDispatcher.php?

Reply:
> Is it possible to do it without wrapping same way as https://github.com/yiisoft/yii-web/blob/master/src/MiddlewareDispatcher.php?
Yes, but the current implementation also has some advantages.
After this PR the handler in the last called middleware will always be an instance of the MiddlewareDispatcher. If a controller action will be the last called middleware as we discussed here https://github.com/yiisoft/yii-web/pull/210, we can add some fallback/additional middleware dynamically after RouterMiddleware (before a NotFoundHandler).

Reply:
> I think it worth adding a phpdoc here describing that the stack holds middleware already wrapped into each other. Currently, based on the name one can assume it's just an array of middleware.

Done.

Reply:
> After this PR the handler in the last called middleware will always be an instance of the MiddlewareDispatcher. 

OK. Is it reflected in tests? It seems to be important design thing.

Reply:
> OK. Is it reflected in tests? It seems to be important design thing.
No, at first we need to deсide about the signature of the Route get(put, post etc) method and methods for adding a middleware.

Reply:
1. A route without middleware may make sense (it could be for creating URLs only) so changing factory methods signature isn't needed.
2. A single method for adding (prepending) a middleware is enough. Choose a name you like.

Reply:
1. It could be like `get(string $pattern, ?MiddlewareInterface = null)`
It is very convenient to set a controller action in the method directly. And very clear.
If you aren't using additional middlewares you can just write 
```php
Route::get('/', new ActionCaller(SiteController::class, 'index', $container));
```
And it is a warranty that the controller action will be processed last. It is important for begginers.
2. I like `addMiddleware(MiddlewareInterface $middleware)`.

Do you approve of removing the to(), then() methods and renaming the prepend() method to addMiddleware()?

Reply:
Check the last version:)

Reply:
1. OK. Makes sense.
2. That's fine. I approve removal and renaming.